user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
I think `403` is wrong because if you have a correct token you are authenticated. `403` suggests to me that even a correct token wouldn't be sufficient.